store/postgres/src/relational.rs

```diff
@@ -283,6 +283,30 @@ impl Layout {
         for table in tables {
             table.as_ddl(&mut out, self)?;
         }
+        // Add the Proof of Indexing table and index. This is not added via
+        // self.tables because that would automatically add some indexes for
+        // some columns that aren't necessary.
+        // TODO: This code is intentionally disabled. The existence of the POI table infers that the POI
+        // is being updated, which is not implemented yet.
+        // Unfinished POI work. See also f7da4ade-f47f-4f5a-a1c6-23ef0885dcaf
+        /*
+        let proof_of_indexing_name = format!("{}.{}", self.schema, "proof_of_indexing_artifacts");
+        writeln!(
+            out,
+            "CREATE TABLE {} (
+                id serial primary key,
+                data_source varchar not null,
+                block_number int not null,
+                artifact bigint not null
+            );",
+            proof_of_indexing_name,
+        )?;
```

Comment:
Not a review, just some thoughts since last we talked about this:

(1) A POI artifact is likely valid for a block range (from the block where the last change happened to just before the block with the next change) That means artifacts should store a `block_range`, and makes them really just another entity type. We should just inject the appropriate `Table` definition into the `Layout`'s list of tables and then manage them with the usual entity methods. 

(2) We should hold off on any marking of artifacts by `data_source` since we don't need it yet. For now, it's enough to have one series of artifacts since each subgraph has a single source of time. We should wait with adding a `data_source` or similar field until we actually implement something where a subgraph has more than one source of time.

Reply:
@lutter: 
Using the entity API turned out [nicely](https://github.com/graphprotocol/graph-node/blob/df2dc6b9bc31cb3246c4d2661e25ea1a0496b20f/core/src/subgraph/instance_manager.rs#L824):

 Is this what you had in mind for determining [whether the POI table exists](https://github.com/graphprotocol/graph-node/blob/df2dc6b9bc31cb3246c4d2661e25ea1a0496b20f/core/src/subgraph/instance_manager.rs#L811)? I'm unclear as to whether adding this to the `Schema` for the side effect of having the tables created and be queryable is the right approach, but didn't see any straightforward alternatives. Where would you put the code that edits the `Schema` to include the additional type definition?

Reply:
> @lutter:
> Using the entity API turned out [nicely](https://github.com/graphprotocol/graph-node/blob/df2dc6b9bc31cb3246c4d2661e25ea1a0496b20f/core/src/subgraph/instance_manager.rs#L824):

Nice! You can't use `$poi` as the entity type though: Postgres names can contain a `$`, but can't start with it [(docs)](https://www.postgresql.org/docs/12/sql-syntax-lexical.html#SQL-SYNTAX-IDENTIFIERS).

> Is this what you had in mind for determining [whether the POI table exists](https://github.com/graphprotocol/graph-node/blob/df2dc6b9bc31cb3246c4d2661e25ea1a0496b20f/core/src/subgraph/instance_manager.rs#L811)? I'm unclear as to whether adding this to the `Schema` for the side effect of having the tables created and be queryable is the right approach, but didn't see any straightforward alternatives. Where would you put the code that edits the `Schema` to include the additional type definition?

What I had in mind there was a check against the actual database schema; externally, that should be a method `Store.supports_poi(&subgraph_id)` which is backed by the same method on `entities::Connection` which in turn always returns `false` for `JsonStorage` and for `Storage::Relational` queries [`information_schema.tables`](https://www.postgresql.org/docs/9.6/infoschema-tables.html) to see if an entry with  `table_name = <name of POI table> and table_schema = <layout.schema>` exists. The result of that query should be cached in the `Layout` and so should probably be passed into `Layout::new` (and `struct Layout` gets a new attribute `supports_poi`)